app/src/main/java/gov/hhs/cdc/trustedintermediary/external/javalin/DomainsRegistration.java

```diff
@@ -111,20 +113,52 @@ static DomainConnector constructNewDomainConnector(Class<? extends DomainConnect
         }
     }
 
-    static Handler createHandler(Function<DomainRequest, DomainResponse> handler) {
+    static Handler createHandler(
+            Function<DomainRequest, DomainResponse> handler, boolean isProtected) {
         return (Context ctx) -> {
             LOGGER.logInfo(ctx.method().name() + " " + ctx.url());
 
             var request = javalinContextToDomainRequest(ctx);
-
-            var response = handler.apply(request);
-
+            DomainResponse response = processRequest(request, handler, isProtected);
             domainResponseFillsInJavalinContext(response, ctx);
 
             LOGGER.logInfo("Handler complete");
         };
     }
 
+    protected static DomainResponse processRequest(
+            DomainRequest request,
+            Function<DomainRequest, DomainResponse> handler,
+            boolean isProtected) {
+        if (isProtected) {
+            DomainResponse authResponse = authenticateRequest(request);
+            if (authResponse != null) {
+                return authResponse;
+            }
+        }
+
+        return handler.apply(request);
+    }
+
+    protected static DomainResponse authenticateRequest(DomainRequest request) {
+        AuthRequestValidator authValidator =
+                ApplicationContext.getImplementation(AuthRequestValidator.class);
+        DomainResponseHelper domainResponseHelper =
+                ApplicationContext.getImplementation(DomainResponseHelper.class);
+        try {
+            if (!authValidator.isValidAuthenticatedRequest(request)) {
+                var errorMessage = "The request failed the authentication check";
+                LOGGER.logError(errorMessage);
+                return domainResponseHelper.constructErrorResponse(401, errorMessage);
+            }
+        } catch (SecretRetrievalException | IllegalArgumentException e) {
+            LOGGER.logFatal("Unable to validate whether the request is authenticated", e);
+            return domainResponseHelper.constructErrorResponse(500, e);
+        }
+
+        return null;
```

Comment:
I'm wondering if we can return a domain request with 200 status code here, and then we check for the status code instead of null in  processRequest; thoughts?

Reply:
I think returning a 200 response at that point may be misleading because we don't know if that's going to be the status code yet. Once we call `handler.apply(request)`, we may need to return a 400 response